cpp/src/arrow/datum.h

```diff
@@ -116,6 +117,27 @@ struct ARROW_EXPORT Datum {
   explicit Datum(std::string value);
   explicit Datum(const char* value);
 
+  // Convenience constructors for a DurationScalar from std::chrono::nanoseconds
```

Comment:
Why put this in Datum instead of Scalar or even DurationScalar?

Reply:
Datum already has many convenience constructors, so it seemed like a logical place. Moreover it enabled my immediate use case of `literal(5min)`. I can also add it to DurationScalar if you like